darts/models/forecasting/foundation_model.py

```diff
@@ -0,0 +1,402 @@
+                # a period of 5 epochs (`patience`)
+                my_stopper = EarlyStopping(
+                    monitor="val_loss",
+                    patience=5,
+                    min_delta=0.05,
+                    mode='min',
+                )
+
+                pl_trainer_kwargs={"callbacks": [my_stopper]}
+            ..
+
+            Note that you can also use a custom PyTorch Lightning Trainer for training and prediction with optional
+            parameter ``trainer`` in :func:`fit()` and :func:`predict()`.
+        show_warnings
+            whether to show warnings raised from PyTorch Lightning. Useful to detect potential issues of
+            your forecasting use case. Default: ``False``.
+        """
+        # initialize `TorchForecastingModel` base class
+        super().__init__(**self._extract_torch_model_params(**self.model_params))
+
+        # extract pytorch lightning module kwargs
+        self.pl_module_params = self._extract_pl_module_params(**self.model_params)
+
+        # validate and set fine-tuning flag
+        raise_if(
+            enable_finetuning and not self.allows_finetuning,
+            f"Fine-tuning is not supported for {self.__class__.__name__}."
+            " Please set `enable_finetuning=False`.",
+            logger,
+        )
+
+        self._enable_finetuning = enable_finetuning
+
+    @classmethod
+    def _validate_model_params(cls, **kwargs):
+        """validate that parameters used at model creation are part of :class:`TorchForecastingModel`,
+        :class:`PLForecastingModule`, :class:`FoundationModel` or cls __init__ methods.
+        """
+        valid_kwargs = (
+            set(inspect.signature(TorchForecastingModel.__init__).parameters.keys())
+            | set(inspect.signature(PLForecastingModule.__init__).parameters.keys())
+            | set(inspect.signature(FoundationModel.__init__).parameters.keys())
+            | set(inspect.signature(cls.__init__).parameters.keys())
+        )
+
+        invalid_kwargs = [kwarg for kwarg in kwargs if kwarg not in valid_kwargs]
+
+        raise_if(
+            len(invalid_kwargs) > 0,
+            f"Invalid model creation parameters. Model `{cls.__name__}` has no args/kwargs `{invalid_kwargs}`",
+            logger=logger,
+        )
+
+    @property
+    def probabilistic(self) -> bool:
+        """Whether the model produces probabilistic forecasts."""
+        return self._probabilistic
+
+    @property
+    def allows_finetuning(self) -> bool:
+        """Whether fine-tuning is allowed for this foundation model."""
+        return self._allows_finetuning
+
+    @property
+    def enable_finetuning(self) -> bool:
+        """Whether fine-tuning is enabled for this foundation model. When enabled, calling `fit()`
+        will update the model weights. When disabled, calling `fit()` will not update the model weights."""
+        return self._enable_finetuning
```

Comment:
I think we can drop these properties and leave them as private attributes (I'm working currently on some Documentation updates, and want to reduce a bit was gets displayed to the user).